### ISSUE TITLE
New feature: configurable mapping from ES6 module names to UI5 import names

### ISSUE THREAD

**Issue by Elberet** [r-murphy/babel-plugin-transform-modules-ui5#16]
I've been thinking about how to handle polyfills in my UI5 application. #14 and #15 are issues I stumbled across during my experiments, but #14 would become irrelevant if this is implemented and #15 is tied to an experimental feature in `@babel/preset-env` and probably not a good long-term solution anyways.

I would like to get this setup to work:
- configure @babel/preset-env with `useBuiltIns: false`,
- add `import "@babel/polyfill"` at the top of `Component.js`,
- copy `@babel/polyfill/dist/browser.min.js` into my project and, finally:
- *let `babel-plugin-transform-modules-ui5` rewrite the canonical polyfill module name into the relative URL of the packaged polyfill library.*

For example:
```js
import "@babel/polyfill"
import UIComponent from "sap/ui/core/UIComponent"

/** @namespace app */
export default class Component extends UIComponent {
  //...
}
```

Desired result:
```js
sap.ui.define(["./vendor/browser-polyfill", "sap/ui/core/UIComponent"], function (__vendor_browser_polyfill, UIComponent) {
  // ...
});
```

To achieve this, the preset could take a new option:
```js
let babelOptions = {
  presets: [
    ["transform-ui5", {
      modulesMap: {
        ["@babel/polyfill"]: "./vendor/browser-polyfill"
      }
    }]
  ]
}
```

Supporting a mapper function would be nice but should not be required due to complications with other packages (e.g. `broccoli-babel-transpiler` requires that the babel options object is serializable).

**Comment by r-murphy** [comment]
Good feature suggestion. Implemented in v7.0.0-rc.5.

**Comment by Elberet** [comment]
Thanks! 😄

**Comment by Elberet** [comment]
Sorry to reopen this, but on second thought, if a function is given for the `modulesMap` option, it should be called with more details beyond the name of the module to be mapped - such as the name of the module/file the plugin is currently processing.

This would allow the mapper function to emit a path relative to the current file's directory, so that I no longer have to worry about the UI5 resource root prefix and where module files are located relative to it.

**Comment by r-murphy** [comment]
This sounds okay to me. What do you think about:

```
modulesMap(src, {
    opts,
    cwd: state.cwd,
    filename: state.filename,
    file: {
          opts: state.file.opts,
    },
})
```

The object passed is a subset of the `state` that babel passes in to the visitor. Alternatively I could pass the whole state, but that would expose some internal variables of the plugin which are subject to change.

Do you think you'd also need the AST node?

**Comment by Elberet** [comment]
I can't come up with a use case where the full AST node would be required to derive the import filename - but that's not to say that someone else couldn't, including myself next month. :wink:

What exactly is `opts` and how is it different from `state.file.opts`?

**Comment by r-murphy** [comment]
Okay, I'll add `node` to the same object.

`opts` are the options provided to this plugin. 
`state.file.opts` is basically the `.babelrc` or `babel.config.js` content plus some additional opts derived by babel.

**Comment by r-murphy** [comment]
Included in 7.0.0-rc.8. Feel free to reopen for any additional props.
